boards/particle_boron/src/main.rs

```diff
@@ -540,7 +542,16 @@ pub unsafe fn start_particle_boron() -> (
     );
     base_peripherals.twi1.set_master_client(i2c_master_slave);
     base_peripherals.twi1.set_slave_client(i2c_master_slave);
-    base_peripherals.twi1.set_speed(nrf52840::i2c::Speed::K400);
+    // Note: strongly suggested to use external pull-ups for higher speeds
+    //       to maintain signal integrity.
+    base_peripherals.twi1.set_speed(nrf52840::i2c::Speed::K100);
+
+    // I2C pin cfg for target
+    _ = &nrf52840_peripherals.gpio_port[I2C_SDA_PIN].set_i2c_pin_cfg();
+    _ = &nrf52840_peripherals.gpio_port[I2C_SCL_PIN].set_i2c_pin_cfg();
+    // Enable internal pull-ups
+    _ = &nrf52840_peripherals.gpio_port[I2C_SDA_PIN].set_floating_state(FloatingState::PullUp);
+    _ = &nrf52840_peripherals.gpio_port[I2C_SCL_PIN].set_floating_state(FloatingState::PullUp);
```

Comment:
```suggestion
    nrf52840_peripherals.gpio_port[I2C_SDA_PIN].set_i2c_pin_cfg();
    nrf52840_peripherals.gpio_port[I2C_SCL_PIN].set_i2c_pin_cfg();
    // Enable internal pull-ups
    nrf52840_peripherals.gpio_port[I2C_SDA_PIN].set_floating_state(FloatingState::PullUp);
    nrf52840_peripherals.gpio_port[I2C_SCL_PIN].set_floating_state(FloatingState::PullUp);
```

Sorry I just recently learned that we don't need all the extra `_ = &` syntax.

Reply:
fixed!